corehq/apps/userreports/expressions/specs.py

```diff
@@ -79,8 +79,18 @@ def configure(self, context):
             raise BadSpecError(u'Name {} not found in list of named expressions!'.format(self.name))
         self._context = context
 
+    def _context_cache_key(self):
+        return 'named_expression-{}'.format(self.name)
```

Comment:
From my understanding named expressions are either the root_doc or the item from `base_item_expression`. and anything inside name expressions will be consistent throughout the doc/iteration as they don't take arguments.

With that said it does feel a little hacky to rely on a second cache, but item can be an arbitrary dictionary which could include something that is not json serializeable (like datetimes) which makes making this vary on item difficult